### ISSUE TITLE
support Namespace and change Web UI to support

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes-sigs/kube-scheduler-simulator#84]
Let's support Namespace. 

I think it requires a big Web UI design change, because all resources are now created on the same namespace. We need to change Web UI to make it easier to understand which namespace each resource resides in.

**Comment by momom-i** [comment]
Hello, again. if no one working on it, I'd like to try it!

For backend side, I have an image for this implement like as follows.
- Add namespace service that has `Get`, `List`, `Apply` and `Delete` method, same as node or pod
- Change some method for namespaced resources to pass namespace info to them

Is my understanding almost correct?

**Comment by sanposhiho** [comment]
Hi @momom-i.
This issue is not started. Please feel free to assign yourself with `/assign`

Yeah, your thought is right, but I feel the development for this issue should be done after #95. 
After #95, we don't need to change the backend and can create/edit/delete namespaces by requesting kube-apiserver directly.

**Comment by momom-i** [comment]
Thank you! I understand, so first I'm trying to work on #95 and then if it works, will be back on this issue later.

**Comment by 196Ikuchil** [comment]
Hi @momom-i✋
Are you currently working on this? 
Would you like me to take it on?

**Comment by momom-i** [comment]
@196Ikuchil Hi! No, I'm not working on it so you can take this freely.

@sanposhiho I've been a bit busy right now, but I'd like to contribute to this project. So I hope I can be back in the near future and we can work together :)

**Comment by 196Ikuchil** [comment]
@momom-i Thanks for the reply👌 Have a good GW.

/assign

**Comment by sanposhiho** [comment]
implementation tasks:
- [ ] Design how to show Pods on namespace
  - I, personally, think it's difficult to show on current Web UI 😓 . And given the current web UI may be deprecated, just add namespace column on table views is enough..?
- [ ] support namespace as other resources.
  - create namespace store, API client functions for namespace... etc

**Comment by 196Ikuchil** [comment]
@sanposhiho 
For about #161. 
Sorry for the lack of explanation.
I'm not talking about `client`.

When we use the importing function, some namespaced resources may belong to various namespaces.
And if we operate these resources (e.g. Delete), we call our API directory from web UI. Therefore, the namespace must also be registered with the API

The importing function call `Apply` method on each ResourceService (e.g. `PodService`).
So, I think that we must do register the namespace if it is new before applying.

**Comment by 196Ikuchil** [comment]
And also there is another bug.
We can't use `metav1.NamespaceAll` in call `DeleteCollection` method. 
```go
if err := s.podService.DeleteCollection(ctx, metav1.NamespaceAll, lopts); err != nil {
    return xerrors.Errorf("failed to delete pods on node %s: %w\n", n.Name, err)
}
```
https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/137/files#diff-375d57e386f20eaa5f09f02bb9d28bfc48ac3dca18d0325f59492208219e5618R112

Therefore, I think it needs to be modified to check all namespaces recursively.
This issue will be resolved in #162.

**Comment by sanposhiho** [comment]
/triage accepted
Not to mark as stale.

**Comment by 196Ikuchil** [comment]
> Design how to show Pods on namespace
I, personally, think it's difficult to show namespace on current Web UI 😓 . And given the current web UI may be deprecated, just add namespace column on table views is enough..?
>
> support namespace as other resources.
create New Namepace button, namespace store, API client functions for namespace... etc
https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/84#issuecomment-1119312105

The first one, I think is related to the table view which is the alpha feature on the web. We wouldn't need to spend any time discussing it if the feature is adopted early.

The second one is a basic function and should be implemented as soon as possible.

**Comment by 196Ikuchil** [comment]
This issue already has been resolved in #236 

/close
